command/agentproxyshared/cache/static_secret_capability_manager.go

```diff
@@ -143,8 +143,10 @@ func (sscm *StaticSecretCapabilityManager) StartRenewingCapabilities(indexToRene
 
 		capabilitiesIndex.IndexLock.RLock()
 		token := capabilitiesIndex.Token
-		indexReadablePathsMap := capabilitiesIndex.ReadablePaths
+		indexReadablePathsMap := map[string]struct{}{}
+		maps.Copy(indexReadablePathsMap, capabilitiesIndex.ReadablePaths)
```

Comment:
I'm a tad confused why this would be better than moving the maps.Keys operation within the lock? You aren't using the copied map later on, and the Keys operation is returning you a distinct/copied slice so that shouldn't be subject to race conditions of `ReadablePaths` later on?

Reply:
There's a read later on to `indexReadablePathsMap`: `if maps.Equal(indexReadablePathsMap, newReadablePaths) {`

Reply:
Ah oops completely missed that.

Reply:
Hehe, so did I, which is why I did a 'take 2' at the fix :P